web/api/v1/api.go

```diff
@@ -464,6 +475,82 @@ func (api *API) queryRange(r *http.Request) (result apiFuncResult) {
 	}, nil, res.Warnings, qry.Close}
 }
 
+func (api *API) queryExemplars(r *http.Request) apiFuncResult {
+	start, err := parseTime(r.FormValue("start"))
+	if err != nil {
+		err = errors.Wrapf(err, "invalid parameter 'start'")
+		return apiFuncResult{nil, &apiError{errorBadData, err}, nil, nil}
+	}
+	end, err := parseTime(r.FormValue("end"))
+	if err != nil {
+		err = errors.Wrapf(err, "invalid parameter 'end'")
+		return apiFuncResult{nil, &apiError{errorBadData, err}, nil, nil}
+	}
+	if end.Before(start) {
+		err := errors.New("end timestamp must not be before start time")
+		return apiFuncResult{nil, &apiError{errorBadData, err}, nil, nil}
+	}
+
+	ctx := r.Context()
+	if to := r.FormValue("timeout"); to != "" {
+		var cancel context.CancelFunc
+		timeout, err := parseDuration(to)
+		if err != nil {
+			err = errors.Wrap(err, "invalid parameter 'timeout'")
+			return apiFuncResult{nil, &apiError{errorBadData, err}, nil, nil}
+		}
+
+		ctx, cancel = context.WithTimeout(ctx, timeout)
+		defer cancel()
+	}
+
+	q, err := api.Queryable.Querier(ctx, timestamp.FromTime(minTime), timestamp.FromTime(maxTime))
+	if err != nil {
+		return apiFuncResult{nil, &apiError{errorBadData, err}, nil, nil}
+	}
+	defer q.Close()
+
+	expr, err := parser.ParseExpr(r.FormValue("query"))
+	if err != nil {
+		return apiFuncResult{nil, &apiError{errorBadData, err}, nil, nil}
+	}
+
+	selectors, err := parser.ExtractSelectors(expr)
```

Comment:
I don't think this is going to produce the desired results, as it only looks as far back as start. Staleness, range vectors, offsets and subqueries all allow for going before that. I'd suggest working off the existing code to cover that.

Reply:
Can you point me in the direction of the relevant portions of the existing query code? Looks like in the engine code this function `findMinTime`?

Reply:
Yes, that's the one.

Reply:
Let me know if how I've updated this function to use `FindMinTime` is acceptable.